File: Algorithm/Programmers/level2_sort_가장큰수.py

'''
정렬 유형

TEST CASE

numbers	return
[6, 10, 2]	"6210"
[3, 30, 34, 5, 9]	"9534330"
[0, 0, 0] "0"
'''


def solution(numbers):
    answer = ''
    if sum(numbers) == 0: return '0'
    
    n_arr = list(map(str,numbers))
    n_arr.sort(key=lambda n: n*3,reverse=True) 

    return ''.join(n_arr)

File: Algorithm/Programmers/java/level2_sort_가장큰수.java
/*
프로그래머스 level2 - sort

numbers	return
[6, 10, 2]	"6210"
[3, 30, 34, 5, 9]	"9534330"

https://school.programmers.co.kr/learn/courses/30/lessons/42746

*/

import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        String answer = "";
        String[] strArray = Arrays.stream(numbers)
            .mapToObj(String::valueOf)
            .toArray(String[]::new);
        Arrays.sort(strArray, (o1,o2) -> (o2+o1).compareTo(o1+o2)); //내림차순
        
        if("0".equals(strArray[0])) return "0"; //전체 0인 경우
        
        for(String s : strArray){
            answer += s;
        }

        return answer;
    }
}
